Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Driver.hs

```diff
@@ -827,7 +827,10 @@ respondQuote (driverId, _) req = do
       case req.response of
         Pulled -> throwError UnexpectedResponseValue
         Accept -> do
-          when (searchReq.autoAssignEnabled == Just True) $ CS.incrementSearchRequestLockCounter searchTryId
+          when (searchReq.autoAssignEnabled == Just True) $ do
+            whenM (CS.isSearchTryCancelled searchTryId) $
```

Comment:
i can see race condition happening here.

Reply:
both `cancelSearch` and `respondQuote` operations can be executed in this implementation.

Reply:
> both `cancelSearch` and `respondQuote` operations can be executed in this implementation.

there was race condition before too. Ok, i'll fix it

Reply:
ohh, i see. sure chances are very less, although if we can fix it properly that's better.